garrysmod/lua/vgui/dscrollpanel.lua

```diff
@@ -27,119 +30,106 @@ function PANEL:Init()
 	self:SetPaintBackgroundEnabled( false )
 	self:SetPaintBorderEnabled( false )
 	self:SetPaintBackground( false )
-
 end
 
 function PANEL:AddItem( pnl )
-
 	pnl:SetParent( self:GetCanvas() )
-
 end
 
 function PANEL:OnChildAdded( child )
-
 	self:AddItem( child )
-
 end
 
 function PANEL:SizeToContents()
-
-	self:SetSize( self.pnlCanvas:GetSize() )
-
+	local canvas = self:GetCanvas()
+	if IsValid( canvas ) then
+		self:SetSize( canvas:GetSize() )
+	end
 end
 
 function PANEL:GetVBar()
-
 	return self.VBar
-
 end
 
 function PANEL:GetCanvas()
-
 	return self.pnlCanvas
-
 end
 
 function PANEL:InnerWidth()
-
 	return self:GetCanvas():GetWide()
-
-end
-
-function PANEL:Rebuild()
-
-	self:GetCanvas():SizeToChildren( false, true )
-
-	-- Although this behaviour isn't exactly implied, center vertically too
-	if ( self.m_bNoSizing && self:GetCanvas():GetTall() < self:GetTall() ) then
-
-		self:GetCanvas():SetPos( 0, ( self:GetTall() - self:GetCanvas():GetTall() ) * 0.5 )
-
-	end
-
 end
 
 function PANEL:OnMouseWheeled( dlta )
-
-	return self.VBar:OnMouseWheeled( dlta )
-
+	local vbar = self:GetVBar()
+	if IsValid( vbar ) then
+		return vbar:OnMouseWheeled( dlta )
+	end
 end
 
 function PANEL:OnVScroll( iOffset )
-
-	self.pnlCanvas:SetPos( 0, iOffset )
-
+	local canvas = self:GetCanvas()
+	if IsValid( canvas ) then
+		canvas:SetPos( 0, iOffset )
+	end
 end
 
 function PANEL:ScrollToChild( panel )
-
 	self:InvalidateLayout( true )
 
-	local x, y = self.pnlCanvas:GetChildPosition( panel )
-	local w, h = panel:GetSize()
-
-	y = y + h * 0.5
-	y = y - self:GetTall() * 0.5
-
-	self.VBar:AnimateTo( y, 0.5, 0, 0.5 )
+	local canvas = self:GetCanvas()
+	if IsValid( canvas ) then

```

Comment:
Again, could be changed to an early return to prevent scope pyramids

Reply:
I don't see anything wrong with additional checks, but given that they weren't there originally, I'll remove them

Reply:
As per my previous message, the problem is not the presence of additional checks, but the fact that they can be implemented in a way that makes the code easier to understand and read.

i.e. Instead of doing this:
```
local canvas = self:GetCanvas()
if IsValid( canvas ) then
	if ( blah ) then
		if ( foo ) then
			-- Code
		end
	end
end
```

You would do this:
```
local canvas = self:GetCanvas()
if ( !IsValid( canvas ) ) then return end

if ( blah ) then
	if ( foo ) then
		-- Code
	end
end
```

Or even this if/when possible
```
local canvas = self:GetCanvas()
if ( !IsValid( canvas ) ) then return end

if ( !blah ) then return end

if ( !foo ) then return end

-- Code
```

Reply:
Yes, alright, I'll keep that in mind.